enclaver/src/run_container.rs

```diff
@@ -0,0 +1,157 @@
+use anyhow::{anyhow, Result};
+use bollard::container::{Config, LogOutput, LogsOptions, WaitContainerOptions};
+use bollard::models::{DeviceMapping, HostConfig, PortBinding, PortMap};
+use bollard::Docker;
+use futures_util::stream::{StreamExt, TryStreamExt};
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::io::AsyncWriteExt;
+
+pub struct RunWrapper {
+    docker: Arc<Docker>,
+    container_id: Option<String>,
+    stream_task: Option<tokio::task::JoinHandle<()>>,
+}
+
+impl RunWrapper {
+    pub fn new() -> Result<Self> {
+        let docker_client = Arc::new(
+            Docker::connect_with_local_defaults()
+                .map_err(|e| anyhow!("connecting to docker: {}", e))?,
+        );
+
+        Ok(Self {
+            docker: docker_client,
+            container_id: None,
+            stream_task: None,
+        })
+    }
+
+    pub async fn run_enclaver_image(
+        &mut self,
```

Comment:
I'm still finding the best patterns myself but for these structs that have a state machine to them, but it might be better to consume `self` and then return a new object. For example:

```
impl RunWrapper {
    pub fn new() -> Result<Self> {
        ...
    }

    pub async fn run_enclave_image(self, ...) -> Result<RunHandle> {
         ...
    }
}

struct RunHandle {
}

impl RunHandle {
    pub async cleanup(self) {
       ...
    }
}
```
This should remove the `Option` fields in your structs since they model "not for this state".

Alternatively you can take in a `CancellationToken` so you end up with:
```
impl RunWrapper {
    pub fn new() -> Result<Self> {
        ...
    }

    pub async fn run_enclave_image(self, token: CancellationToken) -> Result<()> {
         ...
    }
}
```
and no `cleanup` to forget. However you have to make sure you `.await` to completion.